lib/galaxy/webapps/galaxy/api/tools.py

```diff
@@ -551,9 +551,14 @@ def _create(self, trans, payload, **kwd):
         for job in vars.get('jobs', []):
             rval['jobs'].append(self.encode_all_ids(trans, job.to_dict(view='collection'), recursive=True))
             if inputs.get('send_email_notification', False):
-                job_email_action = trans.model.PostJobAction('EmailAction')
-                job.add_post_job_action(job_email_action)
-                new_pja_flush = True
+                # Unless an anonymous user is invoking this via the API it
+                # should never be an option, but check and enforce that here
+                if trans.user is None:
+                    raise exceptions.ToolExecutionError("Anonymously run jobs cannot send an email notification.")
```

Comment:
My gut feeling is that silently ignoring this in the action might be better. If I had to choose between having to resubmit and not getting an email I'd chose the latter.

Reply:
I was torn between this and malformed request.  The only way this should happen is via api submission anonymously, and I think it's probably most appropriate to inform instead of silently no-op.